emsdk.py

```diff
@@ -338,7 +338,13 @@ def win_get_environment_variable(key, system=True):
       # that expansion items such as %PROGRAMFILES% will have been expanded, so
       # need to be precise not to set these back to system registry, or
       # expansion items would be lost.
-      return os.environ[key]
+      if fallback:
+        return os.environ[key]
+      return None
+    finally:
+      if folder is not None:
+        folder.Close()
+
   except Exception as e:
     if e.args[0] != 2:
```

Comment:
I'd love to know under what circumstances these two excpt blocks can fire?   Do we really need two nested try blocks here?  Again, not really related to this PR.

Reply:
Oh, I can clean that up, I think it was there to try to catch exceptions from opening the registry key. I almost did the same in set variable but didn't notice that the get was already doubled up.